js/services/state.ts

```diff
@@ -141,20 +137,7 @@ export const setComponent = (id: string, value: IComponentValue, flowKey: string
 
     components[lookUpKey][id] = Utils.extend(components[lookUpKey][id], value);
 
-    if (value != null) {
-        components[lookUpKey][id].objectData = value.objectData;
-    }
-
-    if (typeof value.isValid === 'undefined' && components[lookUpKey][id].isValid === false) {
-        const model = Model.getComponent(id, flowKey);
-
-        if (model.isRequired &&
-            (!Utils.isNullOrEmpty(value.contentValue as string)  || (value.objectData && value.objectData.length > 0))) {
-
-            components[lookUpKey][id].isValid = true;
-            components[lookUpKey][id].validationMessage = null;
-        }
-    }
+    components[lookUpKey][id] = Utils.extend(components[lookUpKey][id], isValid(id, flowKey));
```

Comment:
Does this line not overwrite the assignment above it?

Reply:
No, it looks strange, but Utils.extend() merges properties. No comments for Utils.extend, but looking at the code, that's what it does,

E.g.

`
var foo = {a:1, b:2}
foo = Utils.extend(foo, {c:true});
{a: 1, b: 2, c: true}
foo = Utils.extend(foo, {c:false});
{a: 1, b: 2, c: false}
foo = Utils.extend(foo, {d:1});
{a: 1, b: 2, c: false, d: 1}
`
When client-side validation is disabled - isValid() always returns true, so then the MOVE request gets to the backend and it does the validation - potentially returning a validation failure.

With client-side validation enabled - Engine.move calls Validation and sets the component state and re-renders if invalid.

Hopefully that makes sense.

Reply:
Ahh, yep OK that makes sense